### DISCUSSION TITLE
ucx lanes select - sys device distance

### DISCUSSION THREAD

**Discussion by EmmaLee-c** [openucx/ucx#9813] [Q&A] [answered]
//MD whose distance is queried when evaluating transport selection score
UCX_SELECT_DISTANCE_MD=cuda_cpy

In this function: 
static void ucp_worker_iface_set_sys_device_distance(ucp_worker_iface_t *wiface)
{
    ......
    for (i = 0; i < context->num_tls; i++) {
        md_index = context->tl_rscs[i].md_index;
        if (strcmp(context->tl_mds[md_index].rsc.md_name,
                   context->config.ext.select_distance_md)) {    // Compare with cuda_cpy
            continue;
        }
        device     = ucp_worker_iface_get_sys_device(wiface);
        cmp_device = context->tl_rscs[i].tl_rsc.sys_device;
        ucs_topo_get_distance(device, cmp_device, distance);
    }
}

Question:
1. What is the function of UCX_SELECT-DISTANCE-MD and why is it fixed as cuda_cpy?
2. Why is it judged to be equal to cuda_cpy in this function, only then will get distance be executed?
3. Do we not need to get distance for other mds, such as mlx5:0?

Thanks very much!

**Comment by yosefe** [comment, marked as answer]
UCX_SELECT_DISTANCE_MD=cuda_cpy means that network devices closest to the currently active cuda device have higher score during transport selection. This configuration specifies memory allocation devices and not network devices, so mlx5_0 is not needed here
